### PR TITLE
Updates to testing and flow to have 100% test coverage

### PR DESCRIPTION
**Pull request by dancoates**
This adds a bunch of stuff to the client generator including:

- enforcing flow annotations in files
- 100% test coverage of template files
- Optional segment tracking implementation
- Working cognito setup
- template files to allow communication with backend

and probably a bunch of other stuff.

### CODE REVIEW

File: generators/client/templates/_webpack.config.js [hunk 1]
```diff
@@ -2,11 +2,13 @@ require('dotenv').config();
 require('babel-register');
 
 var env = [
-    <% if(cognito) { %>'AWS_REGION',
```
Comment (allanhortle) [inline generators/client/templates/_webpack.config.js:5]: Dont remove the if flag here.

File: generators/client/templates/_webpack.config.js [hunk 2]
```diff
@@ -66,6 +68,7 @@ const FILE_LOADER = {
 const development = {
     devtool: 'source-map',
     entry: Object.assign({}, {
+        // 'fetch': 'whatwg-fetch',
```
Comment (dancoates) [inline generators/client/templates/_webpack.config.js:71]: Remove this

File: generators/client/templates/src/__test__/_store-emulate-browser-test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import test from 'ava';
+
+global.window = {
+    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: function(){}
+}
+
+const store = require('../store').default;
+
+test('store', tt => {
```
Comment (allanhortle) [inline generators/client/templates/src/__test__/_store-emulate-browser-test.js]: Maybe replace these assertions with their own tests.

**Comment by dancoates** [comment]
Also add entity constructor stuff:

- https://github.com/blueflag/trc-client-core/blob/develop/src/entity/EntityConstructor.js
- https://github.com/blueflag/trc-toyota-client/blob/develop/client/reducers.js
- https://github.com/blueflag/trc-client-core/blob/develop/src/course/CourseRecord.js
- https://github.com/blueflag/trc-client-core/blob/develop/src/utils/BaseRecord.js

**Review by allanhortle — APPROVED** [review]

**Pull request merged**